src/js/libgeo/GeoOps.js

```diff
@@ -625,6 +625,144 @@ geoOps.PointOnSegment.updatePosition = function(el) {
 };
 geoOps.PointOnSegment.stateSize = 2;
 
+geoOps._helper.PointOnArcCr = function(arc,P){
+    var A = arc.startPoint, B = arc.viaPoint, C = arc.endPoint;
+    //var cr = List.crossratio3harm(P,A,B,C,List.ii);
+    var cr = List.crossratio3harm(A,C,B,P,List.ii);
+    var m = cr.value[0];
+    var n = cr.value[1];
+    if (!CSNumber._helper.isAlmostZero(m)) {
+        n = CSNumber.div(n, m);
+        m = CSNumber.real(1);
+    } else {
+        m = CSNumber.div(m, n);
+        n = CSNumber.real(1);
+    }
```

Comment:
I still don't understand, please explain. `crossratio3harm` returns a homogeneous result, namely a pair [*m* : *n*] which represents the cross ratio *m*/*n*, but includes the special case of infinity if *n* is zero. So *any* multiple of this would represent the same point on the circle. And `normalizeMax` does choose one multiple, in such a way that one of the coordinates becomes 1, just as it is in your code. How can that break anything?